client/my-sites/plans-features-main/index.tsx

```diff
@@ -173,6 +177,58 @@ const SecondaryFormattedHeader = ( { siteSlug }: { siteSlug?: string | null } )
 	);
 };
 
+/**
+ * Helper function to correctly process the plan features for Woo Express plans
+ * with an introductory offer, where we may need to remove some features.
+ */
+const buildPlanFeaturesForWooExpressIntroductoryOffers = (
```

Comment:
Keeping this here adds clutter to the already complicated `plans-features-main` wrapper file. This can be a converted to a hook and live separately, under `hooks` folder. The return value should also be memoized via `useMemo` (from `@wordpress/element`) to ensure it won't cause unnecessary rerenders.

Everything passed as props to `plans-grid` is being memoized in a series of other PRs. see #82460 #82461 #82463 etc.

Reply:
@chriskmnds, I've moved the logic into a separate hook that is memoized at the root of the hook.. Let me know if that was what you were aiming for.

Reply:
Thanks @daledupreez! Nicely done :-)

I'll dismiss my review for now. I can test the PR tomorrow and give it a ✅  if none catches me to it.